hal_st/stm32fxxx/WatchDogStm.cpp

```diff
@@ -2,7 +2,7 @@
 
 namespace hal
 {
-    WatchDogStm::WatchDogStm(const infra::Function<void()>& onExpired)
+    WatchDogStm::WatchDogStm(const infra::Function<void()>& onExpired, Prescalers prescaler)
```

Comment:
I think it would be neat a `struct Config()` with default values
- The struct is to align with other peripherals implementations (like `UartStm`, and `CanStm` which is one that also has a single member in the struct
- A default will keep the new constructor compatible with existing code